### PR TITLE
Add a patch release for stable/lts packages to release.py

### LINKED ISSUE #34741: Add switch for ssh/https protocol for git subcommand in https://github.com/ClickHouse/ClickHouse/blob/0c41564b27478ba2f30d06190035906765493265/tests/ci/release.py

### LINKED ISSUE #34742: Add a `patch AKA stable` release to the script             ClickHouse/tests/ci/release.py                 Line 52      in      0c41564                                         if args.release_type in self.BIG:

### PR DESCRIPTION
**Pull request by Felixoid**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Added patch release for lts/stable versions together with:
- Release branch check helper
- Different protocols for git remote URL
- Improve some edge cases of the release.py

if fixes #34741, fixes #34742

### CODE REVIEW

**Comment by Felixoid** [comment]
style check and python unit test are finished successfully

File: tests/ci/release.py [hunk 1]
```diff
@@ -35,32 +66,45 @@ def run(self, cmd: str, cwd: Optional[str] = None) -> str:
...
+                # Checkout to the commit, it will provide the correct current version
+                if no_prestable:
                     logging.info("Skipping prestable stage")
                 else:
-                    with self.prestable(args):
+                    with self.prestable():
                         logging.info("Prestable part of the releasing is done")
 
-                with self.testing(args):
+                with self.testing():
```
Comment (alesapin) [inline tests/ci/release.py:101]: Looks like we can have both testing and prestable release for a single script run? Testing release shouldn't be `BIG` or `SMALL`...

**Review by alesapin — APPROVED** [review]

File: tests/ci/release.py [hunk 2]
```diff
@@ -35,32 +66,45 @@ def run(self, cmd: str, cwd: Optional[str] = None) -> str:
...
 
+    def set_release_branch(self):
+        # Get the actual version for the commit before check
+        with self._checkout(self.release_commit, True):
+            self.update()
+            self.release_branch = f"{self.version.major}.{self.version.minor}"
+
+        self.update()
+
     def update(self):
```
Comment (alesapin) [inline tests/ci/release.py:77]: Bad name, update what?

File: tests/ci/release.py (hunk 1 above)
Comment (alesapin) [reply tests/ci/release.py:101]: Discussed -- add a comment.

Comment (Felixoid) [reply tests/ci/release.py:101]: it's all ready for a single script running right after the prestable part.

File: tests/ci/release.py [hunk 3]
```diff
@@ -150,35 +240,38 @@ def _bump_prestable_version(self, release_branch: str, args: argparse.Namespace)
         self.run(
             f"git commit -m 'Update version to {new_version.string}' '{cmake_path}'"
         )
-        with self._push(release_branch, args):
+        with self._push(self.release_branch):
             with self._create_gh_label(
-                f"v{release_branch}-must-backport", "10dbed", args
+                f"v{self.release_branch}-must-backport", "10dbed"
```
Comment (alesapin) [inline tests/ci/release.py:245]: add enum for colors

**Review by alesapin — APPROVED** [review]
Another level of CI scripts in ClickHouse...

File: tests/ci/release.py (hunk 3 above)
Comment (Felixoid) [reply tests/ci/release.py:245]: Renamed the argument to `color_hex` and added a comment to `_create_gh_label`

**Comment by Felixoid** [comment]
Style check and python unit tests are done, merging it

**Pull request merged**